parl/remote/scripts.py

```diff
@@ -185,31 +206,43 @@ def stop():
 
 @click.command("status")
 def status():
-    cmd = r'ps -ef | grep remote/monitor.py\ --monitor_port'
-    content = os.popen(cmd).read()
-    pattern = re.compile('--monitor_port (.*?)\n', re.S)
-    monitors = pattern.findall(content)
-    if len(monitors) == 0:
+    cmd = r'ps -ef | grep remote/start.py\ --name\ worker\ --address'
+    content = os.popen(cmd).read().strip()
+    pattern = re.compile('--address (.*?) --cpu')
+    clusters = set(pattern.findall(content))
+    if len(clusters) == 0:
         click.echo('No active cluster is found.')
     else:
         ctx = zmq.Context()
-        status = []
-        for monitor in monitors:
-            monitor_port, _, master_address = monitor.split(' ')
-            monitor_address = "{}:{}".format(get_ip_address(), monitor_port)
-            socket = ctx.socket(zmq.REQ)
-            socket.connect('tcp://{}'.format(master_address))
-            socket.send_multipart([STATUS_TAG])
-            cluster_info = to_str(socket.recv_multipart()[1])
-            msg = """
+        status = []        
+        for cluster in clusters:
+            cmd = r'ps -ef | grep address\ {}'.format(cluster)
+            content = os.popen(cmd).read()
+            pattern = re.compile('--monitor_port (.*?)\n', re.S)
+            monitors = pattern.findall(content)
+            
+            if len(monitors):
+                monitor_port, _, master_address = monitors[0].split(' ')
+                monitor_address = "{}:{}".format(get_ip_address(), monitor_port)
+                socket = ctx.socket(zmq.REQ)
+                socket.connect('tcp://{}'.format(master_address))
+                socket.send_multipart([STATUS_TAG])
```

Comment:
We should add a timeout detection here to avoid that the monitor server exits and this socket keeps waiting.